test_analyzer_streamline_single.py

```diff
@@ -0,0 +1,226 @@
+sNoise = raw_input('Exclude secondaries? ')
+if (hist_dim == '2D'):
+	variable2 = raw_input('Parameter 2: ')
+	y_ax = raw_input('Y axis: ')
+	ybins = int(raw_input('Y bins: '))
+	ylow = float(raw_input('Y low: '))
+	yhigh = float(raw_input('Y high: '))
+
+## initialize root histogram
+hist1 = r.TH1F("2e_tag_" + variable + "_100R_v2","Title;" + x_ax + ";Hits",xbins,xlow,xhigh)
+hist2 = r.TH1F("2e_up_" + variable + "_100R_v2","Title;" + x_ax + ";Hits",xbins,xlow,xhigh)
+hist3 = r.TH1F("2e_down_" + variable + "_100R_v2","Title;" + x_ax + ";Hits",xbins,xlow,xhigh)
+hists = [hist1, hist2, hist3]
+hist4 = r.TH2F("2e_tag_aper_prob_100R_v2","Title;Y (barID);F(Y)",50,0,50,50,0,5)
+hist5 = r.TH2F("2e_up_aper_prob_100R_v2","Title;Y (barID);F(Y)",50,0,50,50,0,5)
+hist6 = r.TH2F("2e_down_aper_prob_100R_v2","Title;Y (barID);F(Y)",50,0,50,50,0,5)
+hists_aper = [hist4, hist5, hist6]
+if (hist_dim == '2D'):
+	hist7 = r.TH2F("2e_tag_" + variable + "_" + variable2 + "_100R_v2","Title;" + x_ax + ";" + y_ax,xbins,xlow,xhigh,ybins,ylow,yhigh)
+	hist8 = r.TH2F("2e_up_" + variable + "_" + variable2 + "_100R_v2","Title;" + x_ax + ";" + y_ax,xbins,xlow,xhigh,ybins,ylow,yhigh)
+	hist9 = r.TH2F("2e_down_" + variable + "_" + variable2 + "_100R_v2","Title;" + x_ax + ";" + y_ax,xbins,xlow,xhigh,ybins,ylow,yhigh)
+	hists = [hist7, hist8, hist9]
+
+#record relative hit efficiency and secondary parameters
+heven_r = [0,0,0]
+hodd_r = [0,0,0]
+heven_f = [0,0,0]
+hodd_f = [0,0,0]
+entries_r = [0,0,0] 	#scale factor is configured in simulation generation script when user determines number of e-/event
+entries_f=[0,0,0]
+
+#cumulative calculations of EEP per module
+#hit_dict_tag = {}
+#hit_dict_up = {}
+#hit_dict_down ={}
+#hit_dicts = [hit_dict_tag, hit_dict_up,hit_dict_down]
+#for i in range(len(hit_dicts)):
+#	for j in range(50):
+#		hit_dicts[i][j] = 0
+
+
+def aper_ratio(x, next_x, prev_x):
+        return (float(next_x + prev_x)/x)
+
+
+## loop over events
+for i in range(cont.tree.numentries):
+    for j in range(len(modules)) :
+    	## get list of pe for tagger array for event i
+    	params = cont.get_data(modules[j],variable,i)
+	if (hist_dim == "2D"):
+		params2 = cont.get_data(modules[j],variable2,i)
+    	barNums=cont.get_data(modules[j],'barID',i)
+    	noiseDiscr = cont.get_data(modules[j],'isNoise',i)
+	rnoiseDiscr = cont.get_data(modules[j],'beamEfrac',i)
+	hit_dict = {}
+	for l in range(50):
+		hit_dict[l] = 0
+    	for k in range(len(params)) :  
+		if ((hist_dim == '1D') and ((pNoise.lower() == 'n') or ((pNoise.lower() == 'y') and (noiseDiscr[k] == False))) and ((sNoise.lower() == 'n') or ((sNoise.lower() == 'y') and ((rnoiseDiscr[k] < 0.2) or (rnoiseDiscr[k] > 0.9)) ))):
+        		hists[j].Fill(params[k])
+		if ((hist_dim == '2D') and ((pNoise.lower() == 'n') or ((pNoise.lower() == 'y') and (noiseDiscr[k] == False))) and ((sNoise.lower() == 'n') or ((sNoise.lower() == 'y') and ((rnoiseDiscr[k] < 0.2) or (rnoiseDiscr[k] > 0.9)) ))):
```

Comment:
it looks like you're actually using an identical treatment for "pure noise" and "secondaries"? but maybe it still makes sense to separate them.

Reply:
Right it is very similar. I justified by this by writing out all the different event makeups (real hit, pure noise, or secondary). I wasn't sure if pure noise events could also have secondaries inside them, so I concluded that I could perform a course cut on pure noise events using isNoise, then a refined cut on secondaries using beamEfrac, and keep them distinct because they are intrinsically different. Also, the secondaries cut, if not coupled with pNoise cuts, will preserve pure Noise (if someone wanted that for whatever reason), because the beamEfrac value of pure noise is < 0.2. I could use beamEfrac values < 0.2 for pure Noise if using isNoise isn't the most robust method and you think that would be better.